parts/k8s/kubernetescustomscript.sh

```diff
@@ -171,12 +170,18 @@ if ! $FULL_INSTALL_REQUIRED; then
     cleanUpContainerImages
 fi
 
+if [ -f /var/run/reboot-required ]; then
+    REBOOTREQUIRED=true
+else
+    REBOOTREQUIRED=false
+fi
+
 if $REBOOTREQUIRED; then
   echo 'reboot required, rebooting node in 1 minute'
-  /bin/bash -c "shutdown -r 1 &"
   if [[ $OS == $UBUNTU_OS_NAME ]]; then
-      aptmarkWALinuxAgent unhold &
+      aptmarkWALinuxAgent unhold
```

Comment:
The reason we were sending this to the background is to accelerate a CSE return code (and thus speed up deployments). Because unholding the walinux mark requires the apt locks to be released, it could take several minutes to reconcile. We don't want to block CSE response on that.

Reply:
There are no more apt locks since I disabled unattended-upgrades and because I decreased to time before the reboot from 1m to 3s I want to make sure this is correctly executed before the reboot occurs.

Reply:
Got it. That makes sense. Except... we've seen instances where cloud-init initiates a background apt operation. That could result in apt locks persisting, whcih might delay apt-mark unhold for a while. Let's put some mileage on this change and compare deployment times before/after. I think the change is ultimately an improvement, just wanna be cautious before introducing additional serial/blocking operations before exiting CSE.

Reply:
In any case we got to unhold before the reboot and like it's the last apt thing before reboot it does not make sens to background it.